server/src/main/java/org/elasticsearch/action/admin/cluster/stats/DenseVectorFieldStats.java

```diff
@@ -31,28 +37,49 @@ public final class DenseVectorFieldStats extends FieldStats {
         indexedVectorCount = 0;
         indexedVectorDimMin = UNSET;
         indexedVectorDimMax = UNSET;
-    }
-
-    DenseVectorFieldStats(StreamInput in) throws IOException {
-        super(in);
-        indexedVectorCount = in.readVInt();
-        indexedVectorDimMin = in.readVInt();
-        indexedVectorDimMax = in.readVInt();
+        vectorIndexTypeCount = new HashMap<>();
+        vectorSimilarityTypeCount = new HashMap<>();
+        vectorElementTypeCount = new HashMap<>();
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
+        assert false : "writeTo should not be called on DenseVectorFieldStats";
         super.writeTo(out);
         out.writeVInt(indexedVectorCount);
         out.writeVInt(indexedVectorDimMin);
         out.writeVInt(indexedVectorDimMax);
+        if (out.getTransportVersion().onOrAfter(DENSE_VECTOR_INDEX_TYPE_TELEMETRY)) {
+            out.writeMap(vectorIndexTypeCount, StreamOutput::writeString, StreamOutput::writeVInt);
+        }
     }
 
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
         builder.field("indexed_vector_count", indexedVectorCount);
         builder.field("indexed_vector_dim_min", indexedVectorDimMin);
         builder.field("indexed_vector_dim_max", indexedVectorDimMax);
+        if (vectorIndexTypeCount.isEmpty() == false) {
+            builder.startObject("vector_index_type_count");
+            for (Map.Entry<String, Integer> entry : vectorIndexTypeCount.entrySet()) {
+                builder.field(entry.getKey(), entry.getValue());
+            }
+            builder.endObject();
+        }
+        if (vectorSimilarityTypeCount.isEmpty() == false) {
+            builder.startObject("vector_similarity_type_count");
+            for (Map.Entry<String, Integer> entry : vectorSimilarityTypeCount.entrySet()) {
```

Comment:
Nit - can't we use `builder.map(vectorSimilarityTypeCount)`?

Reply:
mapContents, but yes, i switched.